File: Know Basic Maths/check_palindrome.cpp
#include<bits/stdc++.h>
using namespace std;

bool check1(string &s){
     int n=s.size();
     string rev(n,' ');
     for(int i=0;i<n;i++){
           rev[i]=s[n-i-1];
     }

     for(int i=0;i<n;i++){
           if(s[i]!=rev[i]){return false;}
     }
     return true;
}

bool check2(string &s){
      int n=s.size();
      for(int i=0;i<n;i++){
            if(s[i]!=s[n-i-1]){return false;}
      }
      return true;
}

int main(){
       string s;
       cout<<"Enter a string "<<endl;
       cin>>s;
       cout<<check1(s)<<endl;
       cout<<check2(s)<<endl;

       // Time complexity :- O(length of string) = O(n)
}

File: Know Basic Maths/check_palindrome.py
def check1(s):
    n = len(s)
    rev = [' '] * n

    for i in range(n):
        rev[i] = s[n - i - 1]

    for i in range(n):
        if s[i] != rev[i]:
            return False
    return True


def check2(s):
    n = len(s)
    for i in range(n):
        if s[i] != s[n - i - 1]:
            return False
    return True


s = input("Enter a string\n")
print(check1(s))
print(check2(s))

# Time complexity: O(length of string) = O(n)
